From 178ae731618d46b5df1ab227dfa0c96920c675df Mon Sep 17 00:00:00 2001
From: Steve Dickson <steved@redhat.com>
Date: Fri, 23 Jan 2015 08:58:05 -0500
Subject: [PATCH] rpc.idmap: Use event_initialized() to check if event is
 initialized

Using EVLIST_INIT directly is bound to break. This seems to be a
leftover from legacy code.

Reported-by: Holger Hoffstatte <holger.hoffstaette@googlemail.com>
Signed-off-by: Steve Dickson <steved@redhat.com>
---
 utils/idmapd/idmapd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/idmapd/idmapd.c b/utils/idmapd/idmapd.c
index c02849b..4246466 100644
--- a/utils/idmapd/idmapd.c
+++ b/utils/idmapd/idmapd.c
@@ -716,7 +716,7 @@ nfsdreopen_one(struct idmap_client *ic)
 		xlog_warn("ReOpening %s", ic->ic_path);
 
 	if ((fd = open(ic->ic_path, O_RDWR, 0)) != -1) {
-		if ((ic->ic_event.ev_flags & EVLIST_INIT))
+		if ((event_initialized(&ic->ic_event)))
 			event_del(&ic->ic_event);
 		if (ic->ic_fd != -1)
 			close(ic->ic_fd);
-- 
1.8.3.1

